deps/build.jl

```diff
@@ -1,68 +1,116 @@
-# Builds GAP with julia GC, compiles/links a few package against it,
-# and exposes the right paths to julia/GAP
+using BinaryProvider
+using Pkg.Artifacts
 
-import Pkg
+# Parse some basic command-line arguments
+const verbose = "--verbose" in ARGS
 
-gap_version = v"4.11.0"
+# Remove deps.jl now, so that we can recognize an aborted build by its absence
+deps_jl = abspath(joinpath(@__DIR__, "deps-$(VERSION).jl"))
+rm(deps_jl, force = true)
 
+
+# TODO: maybe make the artifact loading lazy, for user's who want to use their own GAP???
+# OTOH, these are very few people and I am not sure we should care about them?
+@show artifact_dir = abspath(artifact"gap-4-11-0")
+
+# that artifact directory contains a single subdirectory, with a name like `gap-4.X.Y`;
+# get that name
+content = readdir(artifact_dir)
+@assert length(content) == 1
+gapdirname = content[1]
```

Comment:
maybe matching with regex `r"gap-(4\.\d+\.\d+)"` is better suited here?

Reply:
Either way works; but the actual dirname doesn't really matter, so I decided to instead enforce that there is just one. I could in addition also enforce a format of the dirname, but since this is not actually used later on, I just decided to take whatever is there :-)

Reply:
If there is any chance that the directory `artifact_dir` could collect garbage then it is safer to insist on a specific dirname and even the existence of a specific file in this directory.